src/inmanta/logging.py

```diff
@@ -0,0 +1,249 @@
+"""
+    Copyright 2023 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import logging
+import os
+import sys
+from typing import Optional, TextIO
+
+import colorlog
+from colorlog.formatter import LogColors
+
+from inmanta import const
+
+
+def _is_on_tty() -> bool:
+    return (hasattr(sys.stdout, "isatty") and sys.stdout.isatty()) or const.ENVIRON_FORCE_TTY in os.environ
+
+
+"""
+This dictionary maps the Inmanta log levels to the corresponding Python log levels
+"""
+log_levels = {
+    "0": logging.ERROR,
+    "1": logging.WARNING,
+    "2": logging.INFO,
+    "3": logging.DEBUG,
+    "4": 2,
+    "ERROR": logging.ERROR,
+    "WARNING": logging.WARNING,
+    "INFO": logging.INFO,
+    "DEBUG": logging.DEBUG,
+    "TRACE": 2,
+}
+
+
+class InmantaLogs:
```

Comment:
This class has a bit of an unconventional style, because all the methods are class methods. Normally a singleton contains a class method `get_instance()` that returns the only instance of the class. That way all other methods can be instance methods. This makes the structure of the class more closely related or non-singleton classes and it also makes it clear on the user side that you are applying state changes on the same object. Now, there is just a bunch of calls to static methods on the user side, which makes it harder to interpret what is going on.